gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -133,8 +133,22 @@ private <T> T withRevWalkUnchecked(CheckedFunction1<RevWalk, T> fun) {
     }
   }
 
-  private boolean isBranchPresent(String branchFullName) {
-    return Try.of(() -> jgitRepoForMainGitDir.resolve(branchFullName)).getOrNull() != null;
+  private boolean isBranchPresent(String branchFullName, String currentlyCheckedBranchFullName) {
```

Comment:
`currentlyCheckedBranchFullName` is a veeery confusing name, it looks like _currently checked **out** branch name_, which is definitely isn't supposed to be 😅

Reply:
Also, I wouldn't expose that param in `isBranchPresent`'s interface.. this looks like a detail of recursive implementation which can be hidden from the callers. What callers care for is just whether `branchFullName` exists